mmtk/src/scanning.rs

```diff
@@ -189,6 +198,80 @@ impl Scanning<JuliaVM> for VMScanning {
     }
 }
 
+pub unsafe fn mmtk_scan_gcstack_roots<EV: EdgeVisitor<JuliaVMEdge>>(
```

Comment:
This function looks very similar to the old `mmtk_scan_gcstack`. How is it different? Can you reduce duplicated code between those two functions?

Reply:
The main difference is that `mmtk_scan_gcstack` uses a single closure (as it's called when scanning objects) and `mmtk_scan_gcstack_roots` uses two closures (one for transitively pinned roots, and another for pinning roots). I tried using a single function, but I didn't find any way to call it with a single closure when scanning objects - if I try passing a single closure as both arguments, the rust compiler complains about having two mutable references of the closure.